mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala

```diff
@@ -122,8 +122,8 @@ class GradientDescent private[spark] (private var gradient: Gradient, private va
    * @return solution vector
    */
   @DeveloperApi
-  def optimize(data: RDD[(Double, Vector)], initialWeights: Vector): Vector = {
-    val (weights, _) = GradientDescent.runMiniBatchSGD(
+  def optimize(data: RDD[(Double, Vector)], initialWeights: Vector): (Vector, Double, Integer) = {
```

Comment:
sure, I can add a new method optimizeWithStats. How about if the optimizeWithStats returns an Object OptimizeStats which contains all those 3 fields? @jkbradley 
Later OptimizeStats can be extended ...